File: chrome/android/java/src/org/chromium/chrome/browser/GoogleLocationSettingsHelper.java
// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser;

import org.chromium.base.CalledByNative;

/**
 * Helper interface to read the Google Apps location setting to update the infobar button label.
 * Also, starts the Android Google Apps location settings activity upon request.
 */
public interface GoogleLocationSettingsHelper {

    @CalledByNative
    public String getInfoBarAcceptLabel();

    @CalledByNative
    public boolean isMasterLocationSettingEnabled();

    @CalledByNative
    public boolean isGoogleAppsLocationSettingEnabled();

    @CalledByNative
    public void startGoogleLocationSettingsActivity();
}

File: chrome/browser/android/google_location_settings_helper.cc
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/android/google_location_settings_helper.h"

// Temporary stubs for a 3 way patch

std::string GoogleLocationSettingsHelper::GetAcceptButtonLabel(bool allow) {
  return std::string();
}

bool GoogleLocationSettingsHelper::IsAllowLabel() {
  return false;
}

void GoogleLocationSettingsHelper::ShowGoogleLocationSettings() {
}

bool GoogleLocationSettingsHelper::IsMasterLocationSettingEnabled() {
  return false;
}

bool GoogleLocationSettingsHelper::IsGoogleAppsLocationSettingEnabled() {
  return false;
}

bool GoogleLocationSettingsHelper::IsSystemLocationEnabled() {
  return IsMasterLocationSettingEnabled() &&
      IsGoogleAppsLocationSettingEnabled();
}
